be/src/product/product.service.ts

```diff
@@ -0,0 +1,63 @@
+import { Injectable, NotFoundException } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Connection, Repository } from 'typeorm';
+import { CreateProductInput } from './dto/create-product.input';
+import { UpdateProductInput } from './dto/update-product.input';
+import { Product } from './entities/product.entity';
+
+@Injectable()
+export class ProductService {
+  constructor(
+    @InjectRepository(Product)
+    private productRepository: Repository<Product>,
+    private readonly connection: Connection,
+  ) {}
+
+  create(createProductInput: CreateProductInput): Promise<Product> {
+    const newProductItem = this.productRepository.create(createProductInput);
+    return this.productRepository.save(newProductItem);
+  }
+
+  findAll(): Promise<Product[]> {
+    return this.productRepository.find();
+  }
+
+  async findOne(id: number): Promise<Product> {
+    const product = await this.productRepository.findOne(id);
+    if (!product) throw new NotFoundException(`Product #${id} not found`);
+    return this.productRepository.findOne(id);
+  }
+
+  async update(
+    id: number,
+    updateProductInput: UpdateProductInput,
+  ): Promise<Product> {
+    let product: any = await this.productRepository.findOne(id);
+    if (!product) throw new NotFoundException(`Product #${id} not found`);
+    updateProductInput.categories.forEach(async (id) => {
```

Comment:
This will crash super hard if you have the `updateProductInput.categories` nullable

Reply:
Also this gives 0 visibility/errors when category by that id doesn't exist